app/scripts/controllers/transactions/index.js

```diff
@@ -1996,6 +1996,13 @@ export default class TransactionController extends EventEmitter {
       if (otherTxMeta.id === txId) {
         return;
       }
+      // We don't want to mark as dropped any transactions that are originally created by this account and are incoming
+      if (
+        otherTxMeta.type === TransactionType.incoming &&
+        otherTxMeta.txParams.from === txMeta.txParams.from
```

Comment:
Is this already guaranteed by the `searchCriteria` above since it only finds those with a matching `from`?

Reply:
Right, I removed this.